openlibrary/plugins/upstream/addbook.py

```diff
@@ -779,12 +779,12 @@ def _prevent_ocaid_deletion(self, edition):
 
         # read ocaid from form data
         try:
-            ocaid = [
+            ocaid = next(
                 id['value']
                 for id in edition.get('identifiers', [])
                 if id['name'] == 'ocaid'
-            ][0]
-        except IndexError:
+            )
+        except StopIteration:
             ocaid = None
```

Comment:
Again, I am not sure accepting this suggestion will work because of an issue with deleted lines, but how about passing a default value for when there isn't an identifier such that the `name` is `ocaid`:
```suggestion
        ocaid = next(
            (id['value']
            for id in edition.get('identifiers', [])
            if id['name'] == 'ocaid'),
            None
        )

```

Reply:
Beautiful!!
```
# Give me the first "A" that appears in `s` or else give me "Z".
>>> for s in ("ABC", "BAC", "BCA", "BCD", "CDE", ""):
...     next((char for char in s if char == "A"), "Z")
...
'A'
'A'
'A'
'Z'
'Z'
'Z'
```
We should change [`id` to `id_`](https://stackoverflow.com/questions/77552/id-is-a-bad-variable-name-in-python) for compliance with `ruff rule A001` but otherwise this looks good to me.